specs/dwritecore/DWriteCore.md

```diff
@@ -1235,3 +2038,916 @@ DWRITE_BEGIN_INTERFACE(IDWriteFactory8, "202D44F0-84AC-4D6F-B8D0-704B27675CAA")
+        /// Or it may correspond to a BaseGlyph record defined by COLR version 0.
+        /// </remarks>
+        struct PAINT_LAYERS
+        {
+            /// <summary>
+            /// Number of child paint elements in bottom-up order. Use the IDWritePaintReader
+            /// interface's MoveFirstChild and MoveNextSibling methods to retrieve the child paint
+            /// elements. Use the MoveParent method to return to the parent element.
+            /// </summary>
+            UINT32 childCount;
+        } layers;
+
+        /// <summary>
+        /// Valid for paint elements of type DWRITE_PAINT_TYPE_SOLID_GLYPH.
+        /// Specifies a glyph with a solid color fill.
+        /// This paint element has no child elements.
+        /// </summary>
+        /// <remarks>
+        /// This corresponds to a combination of two paint records in the OpenType COLR table:
+        /// a PaintGlyph record, which references either a PaintSolid or PaintVarSolid record.
+        /// Or it may correspond to a Layer record defined by COLR version 0.
+        /// </remarks>
+        struct PAINT_SOLID_GLYPH
+        {
+            /// <summary>
+            /// Glyph index defining the shape to be filled.
+            /// </summary>
+            UINT32 glyphIndex;
+
+            /// <summary>
+            /// Glyph color used to fill the glyph shape.
+            /// </summary>
+            DWRITE_PAINT_COLOR color;
+        } solidGlyph;
+
+        /// <summary>
+        /// Valid for paint elements of type DWRITE_PAINT_TYPE_SOLID.
+        /// Specifies a solid color used to fill the current shape or clip.
+        /// This paint element has no child elements.
+        /// </summary>
+        /// <remarks>
+        /// This corresponds to a PaintSolid or PaintVarSolid record in the OpenType COLR table.
+        /// </remarks>
+        DWRITE_PAINT_COLOR solid;
+
+        /// <summary>
+        /// Valid for paint elements of type DWRITE_PAINT_TYPE_LINEAR_GRADIENT.
+        /// Specifies a linear gradient used to fill the current shape or clip.
+        /// This paint element has no child elements.
+        /// </summary>
+        /// <remarks>
+        /// This corresponds to a PaintLinearGradient or PaintVarLinearGradient record in the OpenType
+        /// COLR table.
+        /// </remarks>
+        struct PAINT_LINEAR_GRADIENT
+        {
+            /// <summary>
+            /// D2D1_EXTEND_MODE value speciying how colors outside the interval are defined.
+            /// </summary>
+            UINT32 extendMode;
+
+            /// <summary>
+            /// Number of gradient stops. Use the IDWritePaintReader::GetGradientStops method to
+            /// get the gradient stops.
+            /// </summary>
+            UINT32 gradientStopCount;
+
+            /// <summary>
+            /// X coordinate of the start point of the color line.
+            /// </summary>
+            float x0;
```

Comment:
I assume the coordinate system is explained in the OpenType spec, so I won't ask what they mean. I assume that anybody who is working at this low level will have the OpenType spec and be following along.

Reply:
DISCUSSION: Yes, these fields match the names in the OT spec.